client/src/slices/characters/charactersSlice.ts

```diff
@@ -44,6 +44,10 @@ const characterSlice = createSlice({
         state.isLoading = false
         state.currentCharacter = action.payload
       })
+      .addCase('demo/resetState', (state) => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        state = initialState
```

Comment:
same here, we should return it, as this won't work I think (this is true for all slices)

```suggestion
       return initialState
```